easybuild/toolchains/compiler/pgi.py

```diff
@@ -0,0 +1,75 @@
+##
+# Copyright 2015 Bart Oldeman
+#
+# This file is triple-licensed under GPLv2 (see below), MIT, and
+# BSD three-clause licenses.
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Support for PGI compilers (pgcc, pgc++, pgfortran) as toolchain compilers.
+
+@author: Bart Oldeman (McGill University, Calcul Quebec, Compute Canada)
+"""
+
+from distutils.version import LooseVersion
+
+import easybuild.tools.systemtools as systemtools
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.toolchain.compiler import Compiler
+
+
+TC_CONSTANT_PGI = "PGI"
+
+
+class Pgi(Compiler):
+    """PGI compiler class
+    """
+
+    COMPILER_MODULE_NAME = ['pgi']
+
+    COMPILER_FAMILY = TC_CONSTANT_PGI
+
+    # Reference: https://www.pgroup.com/doc/pgiref.pdf
+    COMPILER_UNIQUE_OPTION_MAP = {
+        'i8': 'i8',
+        'r8': 'r8',
+        'optarch': '', # PGI by default generates code for the arch it is running on!
```

Comment:
The target architecture can be specified using the `-tp=<arch>` option.  PGI 14.1 supports the following architectures, newer versions may accept more (can't check atm):
```
    px              Generic x86 Processor
    k8              AMD64 Processor
    k8-64e          AMD64 Processor rev E or later, 64-bit mode
    barcelona       AMD Barcelona processor
    shanghai        AMD Shanghai processor
    istanbul        AMD Istanbul processor
    bulldozer       AMD Bulldozer processor
    piledriver      AMD Piledriver processor
    p7              Intel P7 Architecture (Pentium 4, Xeon, Centrino)
    core2           Intel Core-2 Architecture
    penryn          Intel Penryn Architecture
    nehalem         Intel Nehalem processor
    sandybridge     Intel SandyBridge processor
    haswell         Intel Haswell processor
    x64             Unified AMD/Intel mode; -tp=k8,p7
```